src/aws-cpp-sdk-core/include/smithy/client/AwsSmithyClient.h

```diff
@@ -31,22 +31,19 @@ namespace client
     class AwsSmithyClientT : public AwsSmithyClientBase
     {
     public:
-        AwsSmithyClientT(Aws::Client::ClientConfiguration& clientConfig, const Aws::String& serviceName,
```

Comment:
This should be explicit to avoid usage of default constructor. 
Also, should client be copyable, assignable, moveable?  Interface classes need more scrutiny